utils/network.go

```diff
@@ -24,6 +30,25 @@ func (n Network) URL() string {
 		return "https://alpha4.starknet.io"
 	case MAINNET:
 		return "https://alpha-mainnet.starknet.io"
+	case GOERLI2:
+		return "https://alpha4.starknet.io"
+	case INTEGRATION:
+		return "https://external.integration.starknet.io"
+	default:
+		return ""
+	}
+}
+
+func (n Network) ChainId() string {
```

Comment:
Nice! I just noticed the chain ID is [defined](https://docs.starknet.io/documentation/architecture_and_concepts/Blocks/transactions/#chain-id) to be the `uint` value of the byte-encoded string (`[]byte("SN_MAIN")`). Since this only seems to be used for the transaction hash, we can cheat a little and return a `felt` here.

Reply:
Got you